From cf2dd458f769e3229a8a25d00ee95bf05121de0b Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@freescale.com>
Date: Fri, 11 Dec 2009 15:16:06 +0800
Subject: [PATCH] ENGR00119075 fix iMX23 USB initialization cause wrong power status

USB initialization set power status register to wrong status.
Cause power state change dead loop.
Remove EA code using debug register to control HW_POWER_STS bit.
That is not perfered.

Signed-off-by: Frank Li <Frank.Li@freescale.com>
---
 arch/arm/plat-stmp3xxx/usb_common.c |   15 ---------------
 1 files changed, 0 insertions(+), 15 deletions(-)

diff --git a/arch/arm/plat-stmp3xxx/usb_common.c b/arch/arm/plat-stmp3xxx/usb_common.c
index aac884c..af4e35d 100644
--- a/arch/arm/plat-stmp3xxx/usb_common.c
+++ b/arch/arm/plat-stmp3xxx/usb_common.c
@@ -206,21 +206,6 @@ struct platform_device *host_pdev_register(struct resource *res, int n_res,
 int usb_phy_enable(void)
 {
 	u32 tmp;
-	/*
-	* Set these bits so that we can force the OTG bits high
-	* so the ARC core operates properly
-	*/
-	stmp3xxx_clearl(BM_POWER_CTRL_CLKGATE,
-		      REGS_POWER_BASE + HW_POWER_CTRL);
-	stmp3xxx_setl(BM_POWER_DEBUG_VBUSVALIDPIOLOCK |
-			 BM_POWER_DEBUG_AVALIDPIOLOCK |
-			 BM_POWER_DEBUG_BVALIDPIOLOCK,
-			 REGS_POWER_BASE + HW_POWER_DEBUG);
-	tmp = __raw_readl(REGS_POWER_BASE + HW_POWER_STS);
-	tmp |= BM_POWER_STS_BVALID | BM_POWER_STS_AVALID |
-		       BM_POWER_STS_VBUSVALID;
-	__raw_writel(tmp, REGS_POWER_BASE + HW_POWER_STS);
-
 	/* Reset USBPHY module */
 	stmp3xxx_setl(BM_USBPHY_CTRL_SFTRST,
 		      REGS_USBPHY_BASE + HW_USBPHY_CTRL);
-- 
1.5.4.4

